libs/datamodel/core/src/ast/parser/parse_type_definition.rs

```diff
@@ -0,0 +1,48 @@
+use super::{
+    helpers::{parsing_catch_all, ToIdentifier, Token, TokenExtensions},
+    parse_comments::*,
+    parse_field::parse_field,
+    Rule,
+};
+use crate::ast::*;
+use crate::diagnostics::{DatamodelError, Diagnostics};
+
+pub fn parse_type_definition(token: &Token<'_>, diagnostics: &mut Diagnostics) -> TypeDefinition {
+    let mut name: Option<Identifier> = None;
+    let mut fields: Vec<Field> = vec![];
+    let mut comment: Option<Comment> = None;
+
+    for current in token.relevant_children() {
+        match current.as_rule() {
+            Rule::TYPE_KEYWORD => diagnostics.push_error(DatamodelError::new_legacy_parser_error(
+                "Type declarations have to be indicated with the `type` keyword.",
+                Span::from_pest(current.as_span()),
+            )),
```

Comment:
I don't understand this branch. Should it error?

Reply:
No, that looks incorrect.